girder_worker/app.py

```diff
@@ -232,35 +248,86 @@ def girder_before_task_publish(sender=None, body=None, exchange=None,
                 headers['jobInfoSpec'] = utils.jobInfoSpec(job)
 
             except ImportError:
-                # TODO: Check for self.job_manager to see if we have
-                #       tokens etc to contact girder and create a job model
-                #       we may be in a chain or a chord or some-such
-                pass
+                # Current task will be the parent task in chain case
+                parent_task = app.current_task
+                try:
+                    assert parent_task is not None, 'Parent task is None'
+                    assert parent_task.request is not None, "Parent task's request is None"
+
+                    assert 'girder_api_url' in parent_task.request, \
+                        "Parent task's request does not contain girder_api_url"
+
+                    assert 'girder_client_token' in parent_task.request, \
+                        "Parent task's request does not contain girder_client_token"
+
+                    assert 'id' in parent_task.request, \
+                        "Parent task's request does not contain id"
+
+                    assert 'id' in headers, 'id is not in headers'
```

Comment:
These feel like misuse of `assert`. Can we change these to `if` blocks that raise exceptions?

Reply:
+1

Reply:
Since there was an in person debate about this, aside from being semantically incorrect - Python actually removes assertion statements when optimized, so `assert` definitely shouldn't be used to determine control flow.

Reply:
I changed it to use if statements and it raises a custom exception now. Do you guys think ParentAttributeException is a good name for the exception?